Python/import.c

```diff
@@ -1278,10 +1280,10 @@ PyImport_ImportFrozenModuleObject(PyObject *name)
     if (status == FROZEN_NOT_FOUND || status == FROZEN_DISABLED) {
         return 0;
     }
-    else if (status == FROZEN_BAD_NAME) {
-        return 0;
-    }
     else if (status != FROZEN_OKAY) {
+        if (name == NULL) {
+            name = Py_None;
+        }
```

Comment:
Given the change in behavior (indicate not found -> raise an exception), it is worth adding a comment here about what changed.

```suggestion
        /* FYI, FROZEN_BAD_NAME used to be treated like FROZEN_NOT_FOUND.
            See bpo-45379. */
        if (name == NULL) {
            name = Py_None;
        }
```